### PR TITLE
[OptionsResolver] Documenting Options argument for closure deprecation func

### PR DESCRIPTION
**Pull request by yceruto**
See https://github.com/symfony/symfony/pull/28738

### CODE REVIEW

File: components/options_resolver.rst [hunk 1]
```diff
@@ -660,13 +660,19 @@ This closure is useful to only deprecate some of the allowed types or values of
...
         ->setAllowedTypes('port', array('null', 'int'))
-        ->setDeprecated('port', function ($value) {
+        ->setDeprecated('port', function (Options $options, $value) {
             if (null === $value) {
                 return 'Passing "null" to option "port" is deprecated, pass an integer instead.';
             }
 
+            // deprecation may also depend on another option
+            if ('ssl' === $options['encryption'] && 456 !== $value) {
+                return 'Passing a port different to "456" when the "encryption" option is set to "ssl" is deprecated.';
```
Comment (xabbuh) [inline components/options_resolver.rst]: [...] different port than "456" [...]

**Review by xabbuh — APPROVED** [review]
👍 with a minor tweak

File: components/options_resolver.rst (hunk 1 above)
Comment (yceruto) [reply components/options_resolver.rst]: Fixed.

**Comment by javiereguiluz** [comment]
Thanks Yonel.

**Pull request merged**